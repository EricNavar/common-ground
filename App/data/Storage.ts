import AsyncStorage from '@react-native-async-storage/async-storage';

export const Storage = {
    get: async (key: string) : Promise<string | null> => {
        return await AsyncStorage.getItem(key);
    },
    set: async (key: string, value: string) => {
        await AsyncStorage.setItem(key, value).then(() => {
            console.log('Stored ', value, ' at ', key);
        }).catch((error: any) => {
            console.log('Error storing data at ', key, error);
        });
    },
};