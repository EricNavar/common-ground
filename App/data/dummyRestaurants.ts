export const losPollosHermanos = {
    name: 'Los Pollos Hermanos',
    pic: 'https://static.independent.co.uk/s3fs-public/thumbnails/image/2015/05/01/15/lospolloshermanos.jpg?width=1200',
};

export const tacoSal = {
    name: 'Taco Sal',
    pic: 'https://static.wikia.nocookie.net/breakingbad/images/c/c0/TacoSal.png/revision/latest?cb=20191120130106',
};


export const loyolasFamilyRestaurant = {
    name: 'Loyola\'s Family Restaurant',
    pic: 'https://www.breakingbad-locations.com/wp-content/uploads/2014/09/E06-Cornered.mkv_000921613.jpg',
};
