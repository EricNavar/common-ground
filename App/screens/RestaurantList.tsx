import React from 'react';
import {Restaurant} from '../commonTypes';
import {StyleSheet, ScrollView, TextInput, Text, TouchableOpacity, View} from 'react-native';
import {RestaurantCard} from '../components/RestaurantCard';

const styles = StyleSheet.create({
    restaurantList: {
        paddingTop: 0,
        paddingBottom: 20,
        paddingHorizontal: 10,
    },
    searchBox: {
        padding: 8,
        marginHorizontal: 14,
        marginBottom: 12,
        borderRadius: 6,
        marginTop: 20,
        backgroundColor: '#ddd',
    },
    row: {
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
        marginTop: 8,
    },
    sortButton: {
        padding: 8,
        borderRadius: 24,
        backgroundColor: '#FF6D6E',
        marginLeft: 10,
    },
    sortButtonText: {
        color: 'white',
    },
});

type RestaurantListProps = {
    navigation: {
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        navigate: any;
    };
    route: {
        params: {
            restaurantList: Restaurant[];
        }
    };
    restaurantList: Restaurant[];
};

function RestaurantList(props: RestaurantListProps) {
    const [searchText, setSearchText] = React.useState<string>('');

    const [restaurantList, setRestaurantList] = React.useState<Restaurant[]>(props.route.params.restaurantList);


    if (!restaurantList || restaurantList.length === 0) {
        return null;
    }

    React.useEffect(() => {
        // Force rerender on search to effectively filter through restaurants
    }, [searchText]);

    const onPressFilterPriceButton = () => {
        const temp = restaurantList.sort(function (r1, r2) {
            return r1.priceRating - r2.priceRating;
        });
        console.log(temp);
        setRestaurantList([...temp]);
    };

    const onPressFilterRatingButton = () => {
        const temp = restaurantList.sort(function (r1, r2) {
            return (r1.starRating || 1) - (r2.starRating || 1);
        });
        console.log(temp);
        setRestaurantList([...temp]);
    };

    const onPressFilterDistanceButton = () => {
        const temp = restaurantList.sort(function (r1, r2) {
            return (r1.distanceMiles || 1000) - (r2.distanceMiles || 1000);
        });
        console.log(temp);
        setRestaurantList([...temp]);
    };

    return (
        <ScrollView style={styles.restaurantList}>
            <TextInput
                value={searchText}
                onChangeText={setSearchText}
                placeholder="Search"
                autoCapitalize="words"
                style={styles.searchBox}
            />
            <View style={styles.row}>
                <Text>Sort by:</Text>
                <TouchableOpacity onPress={onPressFilterPriceButton} style={styles.sortButton}>
                    <Text style={styles.sortButtonText}>Price</Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={onPressFilterRatingButton} style={styles.sortButton}>
                    <Text style={styles.sortButtonText}>Rating</Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={onPressFilterDistanceButton} style={styles.sortButton}>
                    <Text style={styles.sortButtonText}>Distance</Text>
                </TouchableOpacity>
            </View>
            {restaurantList.filter(restaurant => restaurant.name.toUpperCase().includes(searchText.toUpperCase())).map((restaurant, index) => (
                <RestaurantCard
                    key={index}
                    restaurant={restaurant}
                    navigation={props.navigation}
                />
            ))}
        </ScrollView>
    );
}

export {RestaurantList};
