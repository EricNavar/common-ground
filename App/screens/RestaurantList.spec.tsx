import { allRestaurants } from '../data/dummyRestaurants';
import { RestaurantList } from './RestaurantList';
import { render } from '@testing-library/react-native';
import React from 'react';

jest.mock(
    'react-native-vector-icons/FontAwesome',
    () => 'MockedFontAwesomeicons'
);

describe('RestaurantList', () => {
    const dummyNavigation = {
        navigate: '',
    };

    it('Renders RestaurantListPage correctly', () => {
        const route = {
            params: {
                restaurantList: allRestaurants,
            },
        };
        const result = render(
            <RestaurantList
                restaurantList={allRestaurants}
                navigation={dummyNavigation}
                route={route}
            />
        );

        expect(result.toJSON()).toMatchSnapshot();
    });

    it('Handles when there are no restaurants', () => {
        const route = {
            params: {
                restaurantList: [],
            },
        };
        const result = render(
            <RestaurantList
                restaurantList={[]}
                navigation={dummyNavigation}
                route={route}
            />
        );

        expect(result.toJSON()).toMatchSnapshot();
    });
});
