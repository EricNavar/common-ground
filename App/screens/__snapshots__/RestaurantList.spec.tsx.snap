// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantList Handles when there are no restaurants 1`] = `null`;

exports[`RestaurantList Renders RestaurantListPage correctly 1`] = `
<RCTScrollView
  style={
    {
      "paddingBottom": 20,
      "paddingHorizontal": 10,
      "paddingTop": 0,
    }
  }
>
  <View>
    <TextInput
      autoCapitalize="words"
      onChangeText={[Function]}
      placeholder="Search"
      style={
        {
          "backgroundColor": "#ddd",
          "borderRadius": 6,
          "marginBottom": 12,
          "marginHorizontal": 14,
          "marginTop": 20,
          "padding": 8,
        }
      }
      value=""
    />
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginTop": 8,
        }
      }
    >
      <Text>
        Sort by:
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#FF6D6E",
            "borderRadius": 24,
            "marginLeft": 10,
            "opacity": 1,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
            }
          }
        >
          Price
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#FF6D6E",
            "borderRadius": 24,
            "marginLeft": 10,
            "opacity": 1,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
            }
          }
        >
          Rating
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#FF6D6E",
            "borderRadius": 24,
            "marginLeft": 10,
            "opacity": 1,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
            }
          }
        >
          Distance
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "margin": 8,
          "maxHWidth": "100%",
          "opacity": 1,
          "paddingHorizontal": 6,
          "paddingVertical": 12,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://static.independent.co.uk/s3fs-public/thumbnails/image/2015/05/01/15/lospolloshermanos.jpg?width=1200",
          }
        }
        style={
          {
            "borderRadius": 4,
            "height": 60,
            "width": 60,
          }
        }
      />
      <View
        style={
          {
            "width": 16,
          }
        }
      />
      <View
        style={
          {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <Text
          style={
            {
              "alignContent": "center",
              "alignItems": "flex-start",
              "display": "flex",
              "marginBottom": 4,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Los Pollos Hermanos
             
          </Text>
          <Text>
            4
          </Text>
          <MockedFontAwesomeicons
            color="orange"
            name="star"
            size={16}
          />
        </Text>
        <Text>
          Fast-food chain serving Mexican-inspired fare such as tacos, quesadillas & nachos.
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#bbb",
          "height": 1,
          "width": "100%",
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "margin": 8,
          "maxHWidth": "100%",
          "opacity": 1,
          "paddingHorizontal": 6,
          "paddingVertical": 12,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://static.wikia.nocookie.net/breakingbad/images/c/c0/TacoSal.png/revision/latest?cb=20191120130106",
          }
        }
        style={
          {
            "borderRadius": 4,
            "height": 60,
            "width": 60,
          }
        }
      />
      <View
        style={
          {
            "width": 16,
          }
        }
      />
      <View
        style={
          {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <Text
          style={
            {
              "alignContent": "center",
              "alignItems": "flex-start",
              "display": "flex",
              "marginBottom": 4,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Taco Sal
             
          </Text>
          <Text>
            3
          </Text>
          <MockedFontAwesomeicons
            color="orange"
            name="star"
            size={16}
          />
        </Text>
        <Text>
          Fast-food chain serving Mexican-inspired fare such as tacos, quesadillas & nachos.
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#bbb",
          "height": 1,
          "width": "100%",
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "margin": 8,
          "maxHWidth": "100%",
          "opacity": 1,
          "paddingHorizontal": 6,
          "paddingVertical": 12,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://www.breakingbad-locations.com/wp-content/uploads/2014/09/E06-Cornered.mkv_000921613.jpg",
          }
        }
        style={
          {
            "borderRadius": 4,
            "height": 60,
            "width": 60,
          }
        }
      />
      <View
        style={
          {
            "width": 16,
          }
        }
      />
      <View
        style={
          {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <Text
          style={
            {
              "alignContent": "center",
              "alignItems": "flex-start",
              "display": "flex",
              "marginBottom": 4,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Loyola's Family Restaurant
             
          </Text>
          <Text>
            4
          </Text>
          <MockedFontAwesomeicons
            color="orange"
            name="star"
            size={16}
          />
        </Text>
        <Text>
          Fast-food chain serving Mexican-inspired fare such as tacos, quesadillas & nachos.
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#bbb",
          "height": 1,
          "width": "100%",
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "margin": 8,
          "maxHWidth": "100%",
          "opacity": 1,
          "paddingHorizontal": 6,
          "paddingVertical": 12,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://static.wikia.nocookie.net/breakingbad/images/c/cc/ElCamino.JPG/revision/latest?cb=20220503235617",
          }
        }
        style={
          {
            "borderRadius": 4,
            "height": 60,
            "width": 60,
          }
        }
      />
      <View
        style={
          {
            "width": 16,
          }
        }
      />
      <View
        style={
          {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <Text
          style={
            {
              "alignContent": "center",
              "alignItems": "flex-start",
              "display": "flex",
              "marginBottom": 4,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            El Camino
             
          </Text>
          <Text>
            5
          </Text>
          <MockedFontAwesomeicons
            color="orange"
            name="star"
            size={16}
          />
        </Text>
        <Text>
          Fast-food chain serving Mexican-inspired fare such as tacos, quesadillas & nachos.
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#bbb",
          "height": 1,
          "width": "100%",
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "margin": 8,
          "maxHWidth": "100%",
          "opacity": 1,
          "paddingHorizontal": 6,
          "paddingVertical": 12,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://www.tacobell.com/medias/blog-2019-new-menu-header.jpg?context=bWFzdGVyfGltYWdlc3wxNjYyNzd8aW1hZ2UvanBlZ3xpbWFnZXMvaDFhL2g5OC84OTI4NTEyMTE0NzE4LmpwZ3xjMjA5YzZjNTQyMmM0MWM1ZWE5N2Q1OGQ0NDJjYWVkMzdiNTE0MjMyMjRhMzhmMTZjNGEwYTc4YmMwOGYwOWYx",
          }
        }
        style={
          {
            "borderRadius": 4,
            "height": 60,
            "width": 60,
          }
        }
      />
      <View
        style={
          {
            "width": 16,
          }
        }
      />
      <View
        style={
          {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <Text
          style={
            {
              "alignContent": "center",
              "alignItems": "flex-start",
              "display": "flex",
              "marginBottom": 4,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Taco Bell
             
          </Text>
          <Text>
            1
          </Text>
          <MockedFontAwesomeicons
            color="orange"
            name="star"
            size={16}
          />
        </Text>
        <Text>
          Fast-food chain serving Mexican-inspired fare such as tacos, quesadillas & nachos.
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#bbb",
          "height": 1,
          "width": "100%",
        }
      }
    />
  </View>
</RCTScrollView>
`;
